lms/djangoapps/instructor_task/tasks_helper/grades.py

```diff
@@ -550,6 +554,75 @@ def _graded_scorable_blocks_to_header(cls, course):
 
 
 class ProblemResponses(object):
+
+    @classmethod
+    def _build_problem_list(cls, course_blocks, root, path=None):
+        """
+        Generate a tuple of display names, block location paths and block keys
+        for all problem blocks under the ``root`` block.
+
+        Arguments:
+            course_blocks (BlockStructureBlockData): Block structure for a course.
+            root (UsageKey): This block and its children will be used to generate
+                the problem list
+            path (List[str]): The list of display names for the parent of root block
+
+        Yields:
+            Tuple[str, List[str], UsageKey]: tuple of a blocks display name, path, and
+                usage key
+        """
+        display_name = course_blocks.get_xblock_field(root, 'display_name')
+        if path is None:
+            path = [display_name]
+
+        yield display_name, path, root
+
+        for block in course_blocks.get_children(root):
+            display_name = course_blocks.get_xblock_field(block, 'display_name')
+            for result in cls._build_problem_list(course_blocks, block, path + [display_name]):
+                yield result
+
+    @classmethod
+    def _build_student_data(cls, user_id, course_id, problem_location):
+        """
+        Generate a list of problem responses for all problem under the
+        ``problem_location`` root.
+
+        Arguments:
+            user_id (int): The user id for the user generating the report
+            course_id (UsageKey): The UsageKey for the course whose report is
```

Comment:
I think this is actually a [`CourseKey` type](https://github.com/edx/opaque-keys/blob/8e0697778a13b8214ef1862e3d91eccc04af1663/opaque_keys/edx/keys.py#L12).

The root XBlock in a course is a `course` block that can be referred to using a `UsageKey`, but generally when referring to a course as a whole one uses `CourseKey`s. For the second parameter, if one wanted to build the report for the whole course, then one would pass in the root block's `UsageKey`.